### ISSUE TITLE
Unable to connect to server with OptiFine on Linux only

### ISSUE THREAD

**Issue by TheRogueGrunt** [sp614x/optifine#3773]
So I have a dual boot Windows 10 and Linux (Pop!_OS, Ubuntu-based) PC and I have Optifine 1.15.2 pre13. On Windows, I can connect to a my friend's server without an issue. However, if I try to connect to that same server with OpitiFine on Linux, I get a generic world. Turning OptiFine off lets me connect like normal on Linux.

**Comment by Brottweiler** [comment]
> if I try to connect to that same server with OpitiFine on Linux, I get a **generic world**.

What does this mean?

**Comment by TheRogueGrunt** [comment]
> > if I try to connect to that same server with OpitiFine on Linux, I get a **generic world**.
> 
> What does this mean?

Rather than loading the world for the server, I get a new standard world like if you just hit new world.

**Comment by Brottweiler** [comment]
Very strange. I run Linux with Minecraft 1.15.2 Optifine pre13 and have never experienced anything like that.

You should probably provide some logfiles.

**Comment by masterX244** [comment]
How are you connecting exactly? there is a Bug related to SRV records and DNS lookup that only triggers on some Linuxes

**Comment by TheRogueGrunt** [comment]
> 

> 

> Very strange. I run Linux with Minecraft 1.15.2 Optifine pre13 and have never experienced anything like that.
> 
> You should probably provide some logfiles.

Where would the log files be? I'll put it on pastebin or something

**Comment by TheRogueGrunt** [comment]
> How are you connecting exactly? there is a Bug related to SRV records and DNS lookup that only triggers on some Linuxes

I've tried both adding the server to the multiplayer list and via direct connect. Its an mcserver if that matters, and Minecraft Realms works without an issue.

**Comment by masterX244** [comment]
Do you connect via a IP address( 4 sets of number separated by dots) or a domain address (text)?
if you use the latter can you post the address used?

**Comment by TheRogueGrunt** [comment]
> Do you connect via a IP address( 4 sets of number separated by dots) or a domain address (text)?
> if you use the latter can you post the address used?

Domain address. Don't wanna post it publicly as we don't have a whitelist setup yet but its [servername].mcserver.id

**Comment by masterX244** [comment]
googled for a different server on the same hoster and that one uses SRV records. Another victim of that bug...
```
 >  nslookup -type=SRV _minecraft._tcp.troop75.mcserver.id
Non-authoritative answer:
_minecraft._tcp.troop75.mcserver.id     service = 10 5 25588 troop75.mcserver.id.

Authoritative answers can be found from:
troop75.mcserver.id     internet address = 192.99.21.188

```

To get the IP address to use run `nslookup -type=SRV _minecraft._tcp.[servername].mcserver.id`

On windows you get a port line that tells the Port to use for connecting.
on Linux you get a line similar to this `_minecraft._tcp.troop75.mcserver.id     service = 10 5 25588 troop75.mcserver.id.`
the third number after `service = ` (in this example the 25588) is the port to use.

For connecting you enter [servername].mcserver.id:[port]

Too bad your friend hasn't set up the whitelist yet (he should do it since any random user could scan for non-protected servers at that hoster) or i could have given you a working address directly

**Comment by TheRogueGrunt** [comment]
> googled for a different server on the same hoster and that one uses SRV records. Another victim of that bug...
> 
> ```
>  >  nslookup -type=SRV _minecraft._tcp.troop75.mcserver.id
> Non-authoritative answer:
> _minecraft._tcp.troop75.mcserver.id     service = 10 5 25588 troop75.mcserver.id.
> 
> Authoritative answers can be found from:
> troop75.mcserver.id     internet address = 192.99.21.188
> ```
> 
> To get the IP address to use run `nslookup -type=SRV _minecraft._tcp.[servername].mcserver.id`
> 
> On windows you get a port line that tells the Port to use for connecting.
> on Linux you get a line similar to this `_minecraft._tcp.troop75.mcserver.id service = 10 5 25588 troop75.mcserver.id.`
> the third number after `service = ` (in this example the 25588) is the port to use.
> 
> For connecting you enter [servername].mcserver.id:[port]
> 
> Too bad your friend hasn't set up the whitelist yet (he should do it since any random user could scan for non-protected servers at that hoster) or i could have given you a working address directly.
> You get a standard world since you end up on a non-whitelisted server at the 25565 port (which is the default for MC). I only know of that bug since a friend had exactly the same issue on a server that i set up and since i know how my stuff is set up it was quickly located (thats why i had that suspicition immediately on this issue)

I'm at work with him atm with my laptop, I can get him to set it up and then share it

**Comment by TheRogueGrunt** [comment]
> googled for a different server on the same hoster and that one uses SRV records. Another victim of that bug...
> 
> ```
>  >  nslookup -type=SRV _minecraft._tcp.troop75.mcserver.id
> Non-authoritative answer:
> _minecraft._tcp.troop75.mcserver.id     service = 10 5 25588 troop75.mcserver.id.
> 
> Authoritative answers can be found from:
> troop75.mcserver.id     internet address = 192.99.21.188
> ```
> 
> To get the IP address to use run `nslookup -type=SRV _minecraft._tcp.[servername].mcserver.id`
> 
> On windows you get a port line that tells the Port to use for connecting.
> on Linux you get a line similar to this `_minecraft._tcp.troop75.mcserver.id service = 10 5 25588 troop75.mcserver.id.`
> the third number after `service = ` (in this example the 25588) is the port to use.
> 
> For connecting you enter [servername].mcserver.id:[port]
> 
> Too bad your friend hasn't set up the whitelist yet (he should do it since any random user could scan for non-protected servers at that hoster) or i could have given you a working address directly.
> You get a standard world since you end up on a non-whitelisted server at the 25565 port (which is the default for MC). I only know of that bug since a friend had exactly the same issue on a server that i set up and since i know how my stuff is set up it was quickly located (thats why i had that suspicition immediately on this issue)

Got whitelist done, its Boatosi.mcserver.id

**Comment by masterX244** [comment]
connect to boatosi..mcserver.id:25577
that should work

**Comment by TheRogueGrunt** [comment]
> connect to boatosi.mcserver.id:25577
> that should work

That did it! I never woukd have thought it was a server bug since it would connect without OptiFine!

**Comment by masterX244** [comment]
Its a optifine bug since optifine ignores the srv DNS records which "redirect" the default port for some reason. It only triggers if the server hoster hides the nonstandard port that way. And that bug somehow only happens under Linux. Windows is not affected

**Comment by wpalmer** [comment]
This bug does not only happen "when hiding a non-standard port", it also triggers when redirecting to a standard port on a different address (eg: foo.com having a web server, but redirecting Minecraft to minecraft.foo.com for convenience)

**Comment by sp614x** [comment]
Probably duplicate of #2917.

**Comment by sp614x** [comment]
Same as #2917.

**Comment by sp614x** [comment]
Fixed in 1.16.2 pre4
